src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1199,6 +1199,47 @@ public void TestConnectionStringEnvVarHasHighestPrecedence()
             }
         }
 
+        /// <summary>
+        /// Test to validate that when an entity which will return a paginated response is queried, and a custom runtime base route is configured in the runtime configuration,
+        /// then the generated nextLink in the response would contain the rest base-route just before the rest path. For the subsequent query, the rest base-route will be trimmed
+        /// by the upstream before the request lands at DAB.
```

Comment:
> For the subsequent query, the rest base-route will be trimmed by the upstream before the request lands at DAB.

I'm not sure the relevance or what you mean by this, did you intend to have a second request and assert that the nextLink is not present?

Reply:
No, this is just an FYI that even though the nextLink generated contains the base-route, it will not be there in the subsequent request uri.

Reply:
I just realized, this assumption may NOT be always true in case DAB is not consumed by SWA. 
```
/// For the subsequent query, the rest base-route will be trimmed
/// by the upstream before the request lands at DAB.
```

What happens if DAB is stand alone hosted, and a base-route gets configured to some NON-default value e.g. the request is sent to:

`https://localhost:5000/base/api/EntityName`

how do we identify the EntityName? Do we error out?
if we are sending the `nextLink` with the complete route, we should be able to handle both the scenarios - a route reaching to us with base-route is still in the path(for non-SWA usecases) as well as a route where base-route is NOT in the path(for SWA use-cases).

Reply:
Good point. I think we need to scope this feature scenario identical to how the SWA experience would work.

1. Using SWA, a client sends a request to `/data-api/*` where `*` is `<rest-api-path>/<entity_name>
    - SWA forwards requests to DAB but does not include `/data-api` in the request, which resulted in the issue being submitted in the first place.
    - This PR allows the nextLink to contain the base-url defined, which when followed by a client, is correctly resolved by dab because it is routed correctly to SWA, and then SWA drops that base-url from the route values that are passed to dab
2. [Your concern] Reverse proxy or Local(non-swa) environment with base route configured: nextLink would contain the base url and following that link would result in an error since DAB doesn't know how to distinguish the base-url from the rest of the route.

This PR should address number 1, but in it's current state does not prevent the error from number 2 from occurring.

Reply:
That's right For #2, we should do one of the following: 
- Prevent this use case altogether for local environment which means - not allowing configuration through dab CLI and restricting its usage to SWA ( which is guaranteed to strip off the base-route before sending it to DAB)
OR 
- Prevent the error from occurring if we enable this for non-swa use cases.

Reply:
With the changes per this: https://github.com/Azure/data-api-builder/pull/1506#issuecomment-1635335758 incorporated, the base-route will only be coming into the picture when DAB is accessed via SWA.